crates/sui-framework/packages/bridge/sources/message.move

```diff
@@ -83,6 +105,42 @@ module bridge::message {
         }
     }
 
+    public fun extract_blocklist_payload(message: &BridgeMessage): Blocklist {
+        // blocklist payload should consist of one byte blocklist type, and list of 33 bytes ecdsa pub keys
+        let payload_length = vector::length(&message.payload);
+        assert!(payload_length == COMPRESSED_ECDSA_PUB_KEY_LENGTH + 1, ETrailingBytes);
+        let bcs = bcs::new(message.payload);
+        let blocklist_type = bcs::peel_u8(&mut bcs);
+        let validator_pub_key = bcs::into_remainder_bytes(bcs);
+        Blocklist {
+            blocklist_type,
+            validator_pub_key
+        }
```

Comment:
does this take care of length of the pub keys?

Reply:
according to your test it looks like it does

Reply:
not magic but bug 🤦🏻‍♂️, the test wasn't testing the extract path.....